certified-connectors/DocuSignDemo/script.csx

```diff
@@ -2118,13 +2118,90 @@ public class Script : ScriptBase
         templateRoles.Add(signer);
         signer = new JObject();
       }
+    }
+
+    newBody["templateRoles"] = templateRoles;
+
+    if (!string.IsNullOrEmpty(query.Get("status")))
+    {
+      newBody["status"] = query.Get("status");
+    }
 
-      if (key.Contains(" Signing Group"))
+    return newBody;
+  }
+
+  private JObject CreateEnvelopeFromTemplateV3BodyTransformation(JObject body)
+  {
+    var templateRoles = new JArray();
+    var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+
+    var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
+    uriBuilder.Path = uriBuilder.Path.Replace("/envelopes/createWithRecipientFields", "/envelopes");
+    this.Context.Request.RequestUri = uriBuilder.Uri;
+
+    var newBody = new JObject()
+    {
+      ["templateId"] = query.Get("templateId"),
+      ["emailSubject"] = query.Get("emailSubject"),
+      ["emailBlurb"] = query.Get("emailBody")
+    };
+
+    Dictionary<string, JObject> recipientMapping = new Dictionary<string, JObject>();
+    foreach (var property in body)
+    {
+      var value = (string)property.Value;
+      var key = (string)property.Key;
+      string[] keyArray = key.Split(new string[]{":::"}, StringSplitOptions.None);
```

Comment:
Instead of identifier names like "Key", "keyArray", I suggest renaming to recipientArray or signersArray. That will make the code more readable.